// laravel-elixir-materialize-css/Config.js
var Elixir = require('laravel-elixir');

/*
 |----------------------------------------------------------------
 | Application Assets
 |----------------------------------------------------------------
 |
 | These properties tell the package where to find the source files 
 | for your application assets. By default these properties will
 | differ to Elixir's configuration, but you may change them.   
 |
 */
var sassPath      = Elixir.config.assetsPath + '/' + Elixir.config.css.sass.folder + '/';
var variablesFile = sassPath + '_variables-materialize.scss';

/*
 |----------------------------------------------------------------
 | Asset Outputs
 |----------------------------------------------------------------
 |
 | Once all of the Materialize files have been merged 
 | and compiled, they'll be saved to your public directory.
 | Here you can choose exactly where they will be saved.
 |
 */
var cssOutputFile      = Elixir.config.publicPath + '/css/materialize.css';
var jsOutputFile       = Elixir.config.publicPath + '/js/materialize.js';
var fontsOutputFolder  = Elixir.config.publicPath + '/fonts';
//var imagesOutputFolder = Elixir.config.publicPath + '/images';

/*
 |----------------------------------------------------------------
 | 'Materialize' Source Location
 |----------------------------------------------------------------
 |
 | The location of 'Materialize' source files is
 | specified here. You really shouldn't have to change
 | this. However, you may change it if you need to.
 |
 */
var matPath     = 'node_modules/materialize-css/';
var matIcoPath     = 'node_modules/mdi/';
var matSass = 	matPath + 'sass/';
var matSassFile = matPath + 'sass/materialize';


/**
 * Export all of the configuration settings set above 
 * so that they will be accessible cleanly as well   
 * as easily through the familiar dot notation.
 */
var config = {
    sassPath                     : sassPath,
    matPath                      : matPath,
    matSass 					: matSass,
    matIcoPath                      : matIcoPath,
    matSassFile                  : matSassFile,
    variablesFile                : variablesFile,
    cssOutputFile                : cssOutputFile,
    jsOutputFile                 : jsOutputFile,
    fontsOutputFolder              : fontsOutputFolder,
    //imagesOutputFolder           : imagesOutputFolder,
};

module.exports = config;
